Normalized capitalization: RATS terms are capitalized, PKI terms are not (#205)
lamps-wg/csr-attestation: draft-ietf-lamps-csr-attestation.md

to be "CSRs". In this document, the terms "CSR" and Certificate Request
message are used interchangeably.

The term "hardware security module (HSM)" is used generically to refer to the
combination of hardware and software designed to protect keys from unauthorized
access. Other commonly used terms include Secure Element and Trusted Execution
Environment.

Since this document is blending terminology from two domains: Remote Attestation, and X.509 PKI, by convention this document presents RATS terminology in upper case such as "Verifier", and X.509 / PKI terminology in lower case such as "certification authority (CA)". This convention should help to disambiguate terms that are common between the two domains; for example a "Verifier" refers to a RATS entity that acts on Evidence, whereas a "verifier" is a PKI entity that acts on certificates.

> This is not to be confused with camel-case words such as "EvidenceStatement" which refer to an ASN.1 type.

Reviewer: ```suggestion
This convention is distinct from camel-case identifiers like "EvidenceStatement", which denote ASN.1 types.
```

In the RATS architecture, when Evidence or an Attestation Result is presented to a Relying Party (RP), the RP may learn detailed information about the Attester unless that information has been redacted or encrypted. Consequently, a certain amount of trust must be placed in the RP, which raises potential privacy concerns because an RP could be used to track devices. This observation is noted in Section 11 of {{RFC9334}}.

> Typically, the RPs considered in the RATS architecture are application services that use remote attestation, rather than PKI-style RAs or CAs. Devices inherently place significant trust in RA/CA infrastructure elements, and therefore any additional information revealed through remote attestation to such entities is generally less concerning than disclosure to application services. The problem of copying Evidence by CAs into an X.509 certificate is discussed in {{sec-con-publishing-x509}}.

Reviewer: ```suggestion
In the RATS architecture, RPs are typically application services that consume remote attestation, rather than PKI-style RAs or CAs. Devices already place substantial trust in RA/CA infrastructure, so additional information disclosed through remote attestation to these entities is generally less sensitive than disclosure to application services. The issue of CAs embedding Evidence into X.509 certificates is discussed in {{sec-con-publishing-x509}}.
```

- Shared Attestation Keys: A manufacturer of devices may provision all devices with the same attestation key(s), or share a common attestation key across devices of the same product family. This approach anonymizes individual devices by making them indistinguishable from others using the same key(s). However, it also means losing the ability to revoke a single attestation key if a specific device is compromised. Care must be taken to avoid embedding uniquely identifying information in the Evidence, as that would reduce the privacy benefits of using remote attestation with shared attestation keys.

- Per-Use Attestation Keys: Devices may be designed to dynamically generate distinct attestation keys (and request the corresponding certificates) for each use case, device, or session. This is analogous to the Privacy CA model, in which a device is initially provisioned with an attestation key and certificate; then, in conjunction with a privacy-preserving CA, it can obtain unique keys and certificates as needed. This strategy reduces the potential for tracking while maintaining strong security assurances. This is the model described in this document.

> - Anonymous Attestation Mechanisms: Direct Anonymous Attestation (DAA) or similar cryptographic methods can be employed to generate blinded attestation signatures. In these schemes, the verifier can validate the attestation using a root key but does not gain a global correlation handle. Thus, repeated use of the same attestation key cannot be exploited to track devices. {{I-D.ietf-rats-daa}} extends the RATS architecture with such a DAA scheme, significantly enhancing privacy.

Reviewer: ```suggestion
- Anonymous Attestation Mechanisms: Direct Anonymous Attestation (DAA) and related cryptographic schemes enable devices to produce attestation signatures that are verifiable against a root key, but unlinkable across different uses. This prevents a verifier from using repeated attestations with the same key as a global correlation handle to track devices. {{I-D.ietf-rats-daa}} extends the RATS architecture with such a DAA scheme, thereby enhancing privacy.
```

## Background Check Model Security Considerations

A PKCS#10 or CRMF certification request typically consists of a
distinguished name, a public key, and optionally a set of attributes,
collectively signed by the entity requesting certification.

> In general usage, since Attestation Key SHOULD be used only for signing attestations,

Reviewer: ```suggestion
In general, because an Attestation Key is intended solely for signing attestations,
```

> the private key used to sign the CSR SHOULD be different from the

Reviewer: ```suggestion
 the private key used to sign a CSR SHOULD be distinct from the
```

> Attesting Key utilized to sign Evidence about the Target

Reviewer: ```suggestion
Attestation Key used to sign Evidence about the Target
```

> Environment, though exceptions MAY be made where CSRs and Evidence are involved in

Reviewer: ```suggestion
 Environment. Exceptions MAY be allowed when CSRs and Evidence are both part of the process
```

> bootstrapping the Attesting Key.

Reviewer: ```suggestion
of bootstrapping the Attestation Key.
```

appropriate and available in-context; thus the issue of freshness is
left up to the discretion of protocol designers and implementers.

> In the case of hardware security modules (HSM), the definition of "fresh" is somewhat ambiguous in the context

Reviewer: ```suggestion
In the case of hardware security modules (HSM), the definition of "freshness" is somewhat ambiguous in the context
```
